p2p-sync/src/main/java/org/smartregister/p2p/authenticator/BaseSyncConnectionAuthenticator.java

```diff
@@ -0,0 +1,47 @@
+package org.smartregister.p2p.authenticator;
+
+import android.support.annotation.NonNull;
+
+import org.smartregister.p2p.contract.P2pModeSelectContract;
+import org.smartregister.p2p.sync.DiscoveredDevice;
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 14/03/2019
+ */
+
+public abstract class BaseSyncConnectionAuthenticator {
+
+    protected P2pModeSelectContract.View view;
+    protected P2pModeSelectContract.Interactor interactor;
```

Comment:
In MVP only the presenter can bind the view and interactor(model).  Try to decouple this class

Reply:
Also the presenter and interactor are not used by the child classes